src/Browser.php

```diff
@@ -335,4 +337,64 @@ public function withOptions(array $options)
 
         return $browser;
     }
+
+    public function withTimeout($timeout)
+    {
+        $browser = clone $this;
```

Comment:
No need to `clone` here, you can rely on the `withOptions()`.

Reply:
Feels a bit meh given you want to remove it, but it works for now and keeps the code simpler